File: 백준/Silver/24479. 알고리즘 수업 － 깊이 우선 탐색 1/알고리즘 수업 － 깊이 우선 탐색 1.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");

const [n, m, r] = input[0].split(" ").map(Number);
let graph = Array.from({length: n+1}, () => []);
let visited = Array.from({length: n+1}, () => false);
let answer = Array.from({length: n+1}, () => 0);
let order = 1;

for (let i=1; i<m+1; i++) {
  const [u, v] = input[i].split(" ").map(Number);
  graph[u].push(v);
  graph[v].push(u);
}

for (let i=1; i<n+1; i++) {
  graph[i].sort((a, b) => a-b);
}

function dfs(x) {
  visited[x] = true;
  answer[x] = order++;

  for (i of graph[x]) {
    if (!visited[i]) dfs(i);
  }
  
  
}

dfs(r);
console.log(answer.slice(1, ).join("\n"));

File: 백준/Silver/24479. 알고리즘 수업 － 깊이 우선 탐색 1/알고리즘 수업 － 깊이 우선 탐색 1.py
import sys 
sys.setrecursionlimit(10**6)
input = sys.stdin.readline

N, M, start = map(int, input().split())
graph = [[] for _ in range(N+1)]
visited = [False]*(N+1)
answer = [0]*(N+1)
order = 1

for _ in range(M):
  u, v= map(int, input().split())
  graph[u].append(v)
  graph[v].append(u)

for list in graph:
  list.sort()


def dfs(v):
  global order
  # print(f'v: {v}')
  visited[v] = True
  answer[v] = order
  order += 1

  for i in graph[v]:
    if not visited[i]:
      # print(f'v, i, order: {v}, {i}, {order}')
      dfs(i)


dfs(start)

for i in range(1, N+1):
  print(answer[i])